app/parse.py

```diff
@@ -8,5 +13,47 @@
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+def parse_courses(course_elements: Tag) -> Course:
+    name = course_elements.select_one("h3").text
+    short_description = course_elements.select_one(
+        "p.mb-32"
+    ).text.strip()
+    duration_elements = course_elements.select(
+        "p[class*=ProfessionCardTags_regularTag]"
+    )
+
+    # Check if duration_elements is not empty before accessing its elements
+    if duration_elements:
+        if len(duration_elements) > 1:
+            duration = duration_elements[1].text.strip()
+        else:
+            duration = duration_elements[0].text.strip()
```

Comment:
The logic assumes that if there are multiple `duration_elements`, the second one is the correct duration. This might not always be true if the webpage structure changes. Consider adding checks or logging to handle unexpected structures.